google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java

```diff
@@ -0,0 +1,543 @@
+            .build();
+    // Create the campaign service.
+    try (CampaignServiceClient campaignServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignServiceClient()) {
+      // Update the campaign.
+      MutateCampaignsResponse response =
+          campaignServiceClient.mutateCampaigns(
+              Long.toString(customerId), ImmutableList.of(operation));
+      // Display the results.
+      for (MutateCampaignResult mutateCampaignResult : response.getResultsList()) {
+        System.out.printf(
+            "Updated campaign with resourceName: %s.%n", mutateCampaignResult.getResourceName());
+      }
+    }
+    return;
+  }
+
+  /**
+   * Returns the DSA settings for a campaign. Throws an error if the campaign does not exists or is
+   * not a DSA campaign.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param campaignId the campaign ID.
+   */
+  private DynamicSearchAdsSetting getDsaSetting(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId) {
+    // Create the query.
+    // You must request all DSA fields in order to update the DSA settings in the following step.
+    String query =
+        "SELECT "
+            + "campaign.id, "
+            + "campaign.name, "
+            + "campaign.dynamic_search_ads_setting.domain_name, "
+            + "campaign.dynamic_search_ads_setting.language_code, "
+            + "campaign.dynamic_search_ads_setting.use_supplied_urls_only "
+            + "FROM "
+            + "campaign "
+            + "WHERE "
+            + "campaign.id = "
+            + campaignId;
+    // Create the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(Long.toString(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Create the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+
+      // Throw an exception a campaign with the provided ID does not exist.
+      if (searchPagedResponse.getPage().getResponse().getTotalResultsCount() == 0) {
+        throw new IllegalArgumentException("No campaign found with ID: " + campaignId);
+      }
+
+      // Get the DSA settings.
+      DynamicSearchAdsSetting dynamicSearchAdsSetting =
+          searchPagedResponse
+              .getPage()
+              .getResponse()
+              .getResults(0)
+              .getCampaign()
+              .getDynamicSearchAdsSetting();
+
+      // Get the campaign domain name.
+      String domainName = dynamicSearchAdsSetting.getDomainName().getValue();
+
+      // Throw an exception if the campaign is not a DSA campaign.
+      if (domainName == null || domainName == "") {
```

Comment:
Instead of checking the value in this way, use `dynamicSearchAdsSetting.hasDomainName()`. The library will never return `null` from `StringValue.getValue()`, but `hasDomainName()` will properly check for null.

Reply:
Done.